webapps/camunda-commons-ui/lib/widgets/variables-table/cam-widget-variables-table.js

```diff
@@ -57,30 +57,54 @@ function noopPromise(info /*, i*/) {
 module.exports = [
   '$uibModal',
   '$translate',
-  function($modal, $translate) {
+  '$document',
+  function($modal, $translate, $document) {
     return {
       template: template,
 
       scope: {
         variables: '=?camVariables',
         headers: '=?camHeaders',
         editable: '=?camEditable',
+        validatable: '@',
 
         isVariableEditable: '=?',
         deleteVar: '=?onDelete',
         saveVar: '=?onSave',
+        onSaved: '=?',
         editVar: '=?onEdit',
         downloadVar: '=?onDownload',
         uploadVar: '=?onUpload',
         onSortChange: '&',
         onValidation: '&',
         onChangeStart: '&',
         onChangeEnd: '&',
+        onToggleEditMode: '=?',
         defaultSort: '=?',
         ignoreTypes: '=?'
       },
 
       link: function($scope) {
+        if ($scope.validatable) {
+          $document.on('click', e => {
+            const modalWindow = angular.element('.modal');
+            const modalBackdrop = angular.element('.modal-backdrop');
+            if (
+              (modalWindow.length === 0 && modalBackdrop.length === 0) ||
+              (modalWindow !== e.target &&
+                modalBackdrop !== e.target &&
+                !modalWindow[0].contains(e.target) &&
+                !modalBackdrop[0].contains(e.target))
+            ) {
+              $scope.$apply(() => {
+                $scope.variables.forEach(
+                  variable => (variable.showFailures = false)
+                );
+              });
+            }
+          });
+        }
```

Comment:
Good catch regarding listener deregistration!

The purpose of this is to hide the failure popovers on document click. The failure popover is shown when adding the variable via the modal dialog. However, the modal dialog is not closed immediately but shows the "confirm" step. With this, I want to prevent that the popover is closed while still interacting with the confirm step of the modal dialog.